src/components/structures/RoomView.js

```diff
@@ -784,29 +794,57 @@ module.exports = createReactClass({
             return;
         }
         const e2eMembers = await room.getEncryptionTargetMembers();
+
+        /*
+        Ensure we trust our own signing key, ie, nobody's used our credentials to
+        replace it and sign all our devices
+        */
+        if (!cli.checkUserTrust(cli.getUserId())) {
+            this.setState({
+                e2eStatus: "warning",
+            });
+            debuglog("e2e status set to warning due to not trusting our own signing key");
+            return;
+        }
```

Comment:
This should all be handled automatically by the js-sdk, so you shouldn't need to do this check here.